### PR TITLE
Refactor test packages

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
Multiple test currently reach across package boundaries to utilise the `GraknTestServer` or the `test-integration/logback.xml`. We can fix this architecture and resolve https://github.com/graknlabs/grakn/issues/5043 at the same time.

## What are the changes implemented in this PR?
* Move `ClientJavaE2E` into client-java repository
* Merge `AttributeUniquenessE2E` into `AttributeUniquenessIT`
* Convert `ConcurrencyE2E` into IT
* Create new package in tests for `test-integration/server/uniqueness` containing `AttributeUniquessIT` and `ConcurrencyIT`

### CODE REVIEW

**Comment by flyingsilverfin** [comment]
@haikalpribadi  im not sure `test-assembly` is the best name for tests like `GraknGraqlCommandsE2E` - that used to be E2E tests. It's a bit awkward since we have `test-assembly-mac-zip` that have avery different meaning. It's more of a `test-assembled-commands` or something

**Comment by haikalpribadi** [comment]
I don't have enough context at the moment to know what is in the actual tests, @flyingsilverfin. But the test terminologies that we have decided on is:
- test-unit: for class level
- test-integration: for package or multi-package level
- test-assembly: for testing the software distribution once you've "assembled" the distribution
- test-deployment: for testing the software distribution once you've assembled and "deployed" the distribution

Ideally, each test should be suffixed with the class, package, or distribution it is testing. Additionally, the specific aspect/behaviour of that distribution you're testing.

Does that help?

**Comment by flyingsilverfin** [comment]
I suppose in that case I need to find a name for something that encompasses "user-experience" or "functionality" in an assembly test

**Comment by haikalpribadi** [comment]
I have a feeling that these test are neither here nor there, and it's just a bunch of cases put together with an scoping domain that it's focusing on. I'd say, if you can migrate each case to BDD scenarios, then we should just dissolve this test?

**Comment by flyingsilverfin** [comment]
Then we'd need behaviour tests that handle high level commands such as runnnig console commands haikal, seems a bit out of scope of the behavior steps we have now? Can discuss on a call or im happy to just keep them as is for now and we'll revisit them again in the future

File: test/assembly/BUILD [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+load("@graknlabs_build_tools//checkstyle:rules.bzl", "checkstyle_test")
+
+java_test(
+    name = "daemon-test-assembly",
```
Comment (haikalpribadi) [inline test/assembly/BUILD:21]: Let's reverse the naming, and have `test` be the prefix. So this one we can call `test-daemon`

File: test/assembly/BUILD [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "//server",
+        "//dependencies/maven/artifacts/commons-io",
+        "//dependencies/maven/artifacts/org/hamcrest:hamcrest-library",
+        "//dependencies/maven/artifacts/org/zeroturnaround:zt-exec",
+        "@graknlabs_client_java//:client-java",
+    ],
+)
+
+java_test(
+    name = "distribution-test-assembly",
```
Comment (haikalpribadi) [inline test/assembly/BUILD:40]: Let's reverse the naming, and have `test` be the prefix. So this one we can call `test-distribution`

**Comment by haikalpribadi** [comment]
Will this PR close #5403 completely, @flyingsilverfin ? The following text in the description will trigger the closing of that issue.

> We can fix this architecture and **resolve #5043** at the same time.

**Comment by flyingsilverfin** [comment]
yes @haikalpribadi it should

**Pull request merged**